### PR TITLE
Lammps dump reader

### LINKED ISSUE #1888: Add support for Lammps dump files
As defined here:

http://lammps.sandia.gov/doc/dump.html

### PR DESCRIPTION
**Pull request by richardjgowers**
Fixes #1888

Changes made in this Pull Request:
 - adds support for lammps dump reader

Still needs docs, tests for a triclinic system

### CODE REVIEW

**Comment by orbeckst** [comment]
Btw, we can't merge this into a 0.18.1 because it is a new feature. 

Do you want to hammer out a quick 0.18.1 and then move forward to 0.19.0 or 1.0.0?

File: package/MDAnalysis/coordinates/LAMMPS.py [hunk 1]
```diff
@@ -443,3 +452,126 @@ def write(self, selection, frame=None):
...
+
+class DumpReader(base.ReaderBase):
+    """Reads the default `LAMMPS dump format`_
+
+    Expects trajectories produced by the default 'atom' style dump.
+
+    Will automatically convert positions from their scaled/fractional
+    representation to their real values.
+
+    .. versionadded:: 0.18.1
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/LAMMPS.py]: Has to be 0.19.0 because it is a new feature.

File: package/MDAnalysis/coordinates/LAMMPS.py [hunk 2]
```diff
@@ -443,3 +452,126 @@ def write(self, selection, frame=None):
...
+                line = f.readline()
+                counter += 1
+        self._offsets = offsets[:-1]  # last is EOF
+        return len(self._offsets)
+
+    def close(self):
+        self._file.close()
+
+    def _read_frame(self, frame):
+        self._file.seek(self._offsets[frame])
```
Comment (orbeckst) [inline package/MDAnalysis/coordinates/LAMMPS.py]: Does `seek` work on compressed files?

**Comment by richardjgowers** [comment]
@orbeckst yeah one day I'll manage to get a patch release out.  There's no point holding this back just so we can have an X.1 release, if this gets finished and merged it'll force 19.0 I guess.  We should have a release soon

File: package/MDAnalysis/topology/LAMMPSParser.py [hunk 3]
```diff
@@ -575,6 +580,54 @@ def _parse_box(self, header):
         return unitcell
 
 
+class LammpsDumpParser(TopologyReaderBase):
+    """Parses Lammps ascii dump files in 'atom' format
+
+    Only reads atom ids.  Sets all masses to 1.0.
+
+    .. versionadded:: 0.18.1
```
Comment (orbeckst) [inline package/MDAnalysis/topology/LAMMPSParser.py]: change to 0.19.0

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 4]
```diff
@@ -393,3 +395,81 @@ def test_write_non_numerical_types(self, tmpout):
...
+    def test_length(self, u):
+        assert len(u.trajectory) == 3
+        for i, ts in enumerate(u.trajectory):
+            assert ts.frame == i
+            assert ts.data['step'] == i * 500
+        for i, ts in enumerate(u.trajectory):
+            assert ts.frame == i
+            assert ts.data['step'] == i * 500
+
+    def test_seeking(self, u, reference_positions):
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: Please also test compressed files: zip, gzip, bz2

File: testsuite/MDAnalysisTests/coordinates/test_lammps.py [hunk 5]
```diff
@@ -393,3 +395,81 @@ def test_write_non_numerical_types(self, tmpout):
             assert 'must be convertible to integers' in e.args[0]
         else:
             raise pytest.fail()
+
+
+class TestLammpsDumpReader(object):
+    @pytest.fixture()
```
Comment (orbeckst) [inline testsuite/MDAnalysisTests/coordinates/test_lammps.py]: You could save your data file as bz2 compressed and then create fixtures that create the uncompressed, gz, and zip versions for the test.

File: package/MDAnalysis/coordinates/LAMMPS.py (hunk 2 above)
Comment (richardjgowers) [reply package/MDAnalysis/coordinates/LAMMPS.py]: Apparently yes, we test seeking and reading which relies on this

**Comment by orbeckst** [comment]
Can you please fix the docs https://travis-ci.com/MDAnalysis/mdanalysis/jobs/123399600 and the linter is also unhappy.

**Comment by orbeckst** [comment]
@richardjgowers before you merge this, consider if you don't want to push out a quick 0.18.1 – we had two users already complaining about #1872 . If we move to 0.19.0 then invariably it will take much longer to get a new release out.

**Comment by richardjgowers** [comment]
Releasing a 19.0 is the same effort as a 18.1. No reason why we can’t do a
skinny 19.0 / no reason in holding back features.

**Comment by orbeckst** [comment]
Fair enough.

**Comment by richardjgowers** [comment]
I agree that we should release soon though

**Pull request merged**